docs/SIL.rst

```diff
@@ -6030,9 +6038,9 @@ end_apply
 `````````
 ::
 
-  sil-instruction ::= 'end_apply' sil-value
+  sil-instruction ::= 'end_apply' sil-value ':' sil-type
```

Comment:
This is unfortunate syntax; it makes it look like the token value is being declared to have that type. I would suggest a keyword (e.g. `end_apply %token as $Int`).  It's too bad we can't just recover it from the `begin_apply`, but we can't rely on that instruction preceding this one in the printed SIL.

The SIL verifier needs to enforce the rule that this type matches the function type of the `begin_apply`.